packages/eventstream-codec/src/Int64.ts

```diff
@@ -0,0 +1,65 @@
+import { Int64 as IInt64 } from "@aws-sdk/types";
+import { toHex } from "@aws-sdk/util-hex-encoding";
+
+export interface Int64 extends IInt64 {}
+
+/**
+ * A lossless representation of a signed, 64-bit integer. Instances of this
+ * class may be used in arithmetic expressions as if they were numeric
+ * primitives, but the binary representation will be preserved unchanged as the
+ * `bytes` property of the object. The bytes should be encoded as big-endian,
+ * two's complement integers.
+ */
+export class Int64 {
+  constructor(readonly bytes: Uint8Array) {
+    if (bytes.byteLength !== 8) {
+      throw new Error("Int64 buffers must be exactly 8 bytes");
+    }
+  }
+
+  static fromNumber(number: number): Int64 {
+    // eslint-disable-next-line @typescript-eslint/no-loss-of-precision
+    if (number > 9223372036854775807 || number < -9223372036854775808) {
```

Comment:
is it a problem if a slightly larger number is allowed through? the numbers don't evaluate precisely, and the following will be `false`

```
9_223_372_036_854_776_000 > 
9_223_372_036_854_775_807
```

Reply:
This is copied from https://github.com/aws/aws-sdk-js-v3/blob/52f2b2609c9277030c5de0eeade8df9c5d701c17/packages/eventstream-marshaller/src/Int64.ts#L20-L22

I created a bug report at [ToDo: share GitHub issue]